server/src/api/modules/players/services/UpdatePlayerService.ts

```diff
@@ -26,10 +26,18 @@ async function updatePlayer(username: string, skipFlagChecks = false): Promise<U
   // Find a player with the given username or create a new one if needed
   const [player, isNew] = await findOrCreate(username);
 
+  if (player.annotations?.some(a => a.type === PlayerAnnotationType.OPT_OUT)) {
+    throw new ForbiddenError('Failed to update: Player has opted out.');
```

Comment:
...opted out of tracking.

Reply:
Maybe include a version of `If this is your account and you want to opt back in, contact us on Discord` in this. Incase of double name changes or name changes that are not properly tracked.

Reply:
👍 nice, yep